### PR TITLE
[CPU] Add DFT/IDFT ops

### PR DESCRIPTION
**Pull request by apertovs**
Tickets: 49359, 49375

### CODE REVIEW

**Review by nshchego — APPROVED** [review]

**Comment by apertovs** [comment]
@dmitry-gorokhov, have a look, please.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_dft_node.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    void getSupportedDescriptors() override;
+    void initSupportedPrimitiveDescriptors() override;
+    void createPrimitive() override;
+    void execute(mkldnn::stream strm) override;
+    bool created() const override;
+
+    static bool isSupportedOperation(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept;
+
+private:
+    void execDFT() const;
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_dft_node.h:27]: Just wondering should the all methods below be presented in DFT class or they can be moved in cpp into anonymous namespace (for example then they don't use internal fields/methods)?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/dft.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    ::testing::ValuesIn(axes1D),
+    ::testing::ValuesIn(signalSizes1D),
+    ::testing::ValuesIn(opTypes),
+    ::testing::Values(CommonTestUtils::DEVICE_CPU)
+);
+
+/* 2D DFT */
+
+const std::vector<std::vector<int64_t>> axes2D = {
+    {0, 1}, {1, 2}, {2, 3}, {0, 2}, {1, 3}
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/dft.cpp:49]: negative axes are not covered

**Comment by dmitry-gorokhov** [comment]
@iefode Could you please review the tests part?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp [hunk 3]
```diff
@@ -65,6 +65,9 @@ std::vector<std::string> disabledTestPatterns() {
         // Unsupported operation of type: NormalizeL2 name : Doesn't support reduction axes: (2.2)
         R"(.*BF16NetworkRestore1.*)",
         R"(.*MobileNet_ssd_with_branching.*)",
+
+        // incorrect reference implementation
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp:69]: Please put the ticket number in the comment

**Comment by apertovs** [comment]
@vgavrilo, please, have a look at python test_ngraph part.

**Review by vgavrilo — APPROVED** [review]

**Comment by vgavrilo** [comment]
Fixes in nGraph Python API tests for DFT and IDFT are good for me.

**Review by iefode — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
@apertovs please rebase on latest master

**Review by ilyachur — APPROVED** [review]
nGraph part LGTM 😄

**Pull request merged**